src/main/java/frc/robot/ShooterAuton.java

```diff
@@ -15,10 +15,10 @@
     private WPI_TalonSRX hopper;
     private WPI_TalonSRX ejector;
     private ArrayList<ShotChoice> choices;
-
+    
     public ShooterAuton() {
         holo = HoloTable.getInstance();
-        driveTrain = new DriveTrain();
+        driveTrain = Robot.driveTrain;
```

Comment:
Any chance you can make driveTrain a private variable with a public accessor? Thanks!

Reply:
Personally, I hate public accessors, variables can just be public, but I get it is Java style.  

For this year, I'm actually going to push for a different design strategy (6861 code did this), where the constructor requires all the subsystems it is going to modify (so like the Auton class would have constructor parameters of gyro, drivetrain, etc).  This way, the classes are much easier to test, as we can inject dummy classes in of the same interface to manipulate during testing.

I am drafting this change though for a public accessor.

Reply:
I know accessors are a pain, but yeah. That's Java. I'm open to new ideas when it comes to design patterns. We can discuss it in more detail at some point.